### PR TITLE
turns 304 lines of snowflake code into 24 lines

### PR DESCRIPTION
**Pull request by timothyteakettle**
## About The Pull Request
condenses some awful snowflake code
**testmerge this first please incase something goes fucky** 
I have tested it, but because it deletes a large portion of snowflake code to refactor it into something smaller and more manageable, it's probably best it's testmerged first for a few rounds to make sure everything is fine

also it unlocks horns and meat types so they can be used regardless of species

i want to make all tails into one type, but i don't want to touch too much of this stuff in one pr, so i'll probably do it when/if this is merged

## Why It's Good For The Game
shitcode is bad

## Changelog
:cl:
refactor: snowflake code for mutant bodypart selection has been rewritten to be ~14x shorter
tweak: meat type and horns can now be selected by any species
/:cl:

### CODE REVIEW

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -1919,7 +1629,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 							features["tail_human"] = "None"
 							features["tail_lizard"] = "None"
 
-				if("meats")
+				if("_type")
```
Comment (Kraseo) [inline code/modules/client/preferences.dm]:
```suggestion
				if("meat_type")
```

Comment (timothyteakettle) [reply code/modules/client/preferences.dm]: woops didn't notice i did that, thanks

**Comment by timothyteakettle** [comment]
everything is functioning aside from unlocked bodyparts which are intended to be shown to all species, which i'm working on now

**Comment by timothyteakettle** [comment]
works now 
edit: fuck there's still some weird stuff after more extensive testing
double edit: the linter failing had the solution, it works now, you can pick and wear all the stuff

**Comment by Poojawa** [comment]
Oh good.

I fucking hated this UI code. So So So SO much.

**Comment by timothyteakettle** [comment]
want to make some changes so meat type is only set for anthros but can't make the change on my phone so hey don't merge this yet

edit: done its all cool now

**Comment by silicons** [comment]
testmerge soon

**Pull request merged**